ci_framework/roles/devscripts/tasks/00_pre_check.yml

```diff
@@ -0,0 +1,65 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Check for existing deployment
+
+
+- name: Gather dev-scripts deployed cluster auth information.
+  ansible.builtin.stat:
+    path: >-
+      {{
+        [
+          cifmw_devscripts_repo_dir,
+          'ocp',
+          cifmw_devscripts_config['cluster_name'],
+          'auth',
+          'kubeadmin-password'
+        ] | ansible.builtin.path_join
+      }}
+  register: _kubeconfig_result
+
+- name: Verify the deployed cluster.
+  when:
+    - _kubeconfig_result.stat is defined
+    - _kubeconfig_result.stat.exists
+  block:
+    - name: Gather the deployed cluster authentication information.
+      ansible.builtin.import_tasks: sub_tasks/51_set_facts.yml
+
+    - name: Login to the deployed OpenShift cluster.
+      kubernetes.core.k8s_auth:
+        host: "{{ cifmw_openshift_api }}"
+        password: "{{ cifmw_openshift_password }}"
+        state: present
+        validate_certs: false
+      register: _login_result
+
+    - name: Authentication successful to existing cluster.
+      when:
+        - _login_result is not failed
+        - _login_result.k8s_auth.api_key is defined
+      ansible.builtin.set_fact:
+        cifmw_devscripts_skip_deploy: true
+
+- name: Cleanup the existing environment.
```

Comment:
this will create an issue in the way the reproducer is working: since we're using overlays, the cluster will be down - so basically we'll hit that path every single time.

Reply:
It seems i'm missing something, but shouldn't there be active OCP instance (regardless if overlay based or not) ?

I mean for the case of reuse:
- are there creds to working ocp? ==> reuse
- else (creds not available or not working) ==> need new deployment (might need cleanup [but not bottom layers]? can deploy faster utilizing layers or needs full slow deploy? ...

Reply:
> It seems i'm missing something, but shouldn't there be active OCP instance (regardless if overlay based or not) ?
> 
> I mean for the case of reuse:
> 
>     * are there creds to working ocp? ==> reuse
> 
>     * else (creds not available or not working) ==> need new deployment (might need cleanup [but not bottom layers]? can deploy faster utilizing layers or needs full slow deploy? ...

Sorry, used wrong account by mistake!
~~ Q.

Reply:
Maybe to clarify...
Would the issue be due as per @cjeanner's comment, that there not being active cluster due to overlays? But after successful (at least partial) previous execution there has to be some working OCP (overlay or not), correct?
Or the issue would be with invoking the cleanup here which would cause issue to overlay approach?

Reply:
Reproducer works as follow:
- deploy OCP using devscripts role
- stop OCP cluster (via a specific path in the devscripts role)
- create overlays
- start overlays

Now, we clean using `ansible-playbook reproducer-clean.yml` without the deepscrub. In this state, we:
- still have the devscripts qcow2 on disk
- don't have any running VM
and if we re-run the reproducer, we'll hit that new path introduced by @psathyan - but it will fail the test, and clean the deployed env, and.... we'll have to start over, instead of just getting our overlays.

Reply:
Thanks @cjeanner for the reproducer workflow. It improved my understand of reproducer.

In the latest patch, a check exists for `ocp_*.yml` in the `working_dir` before proceeding with the flow. This file pattern is created only when an earlier run had `cifmw_devscripts_use_layers` set to `true`.

Reply:
With the second run, we can avoid the `devscripts_layout`. For example
```
ansible-playbook -e @scenarios/reproducers/validated-architecture-1.yml \
    -e @~/configs/validated-architecture-1.yml 
    reproducer.yml
    --skip-tags packages
```